CMakeLists.txt

```diff
@@ -18,30 +18,35 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with libVersioningCompiler. If not, see <http://www.gnu.org/licenses/>
 
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.20)
 
-project(libVersioningCompiler)
+project(
+  libVersioningCompiler
+  VERSION 2.0
+  DESCRIPTION
+    "library for runtime compilation and dynamic loading of a C/C++ function"
+  LANGUAGES CXX C)
 
 #----- Set the name of the Version Compiler library
-set (VC_LIB_NAME "VersioningCompiler")
+set(VC_LIB_NAME "VersioningCompiler")
 
-###############################################
-##       Set the library directory           ##
-###############################################
+#############################################
+#       Set the library directory           #
+#############################################
 
-link_directories( ${CMAKE_CURRENT_BINARY_DIR} )
+link_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-###############################################
-##             EXTERNAL LIB                  ##
-###############################################
+#############################################
+#             EXTERNAL LIB                  #
+#############################################
 
 # Load additional CMake modules
-list (APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-# DL library
-find_package(DL REQUIRED)
-include_directories(${DL_INCLUDE_DIR})
-link_directories(${DL_LIBRARY})
+# check for DL library
+if(NOT CMAKE_DL_LIBS)
+  message(FATAL_ERROR "Missing DL libs!")
+endif(NOT CMAKE_DL_LIBS)
```

Comment:
This check always fails on macOS because the dynamic linker is always used and libdl.so is not used/needed. You need to explicitly disable the check on Apple platforms like this:
```
if(NOT ${APPLE} AND NOT CMAKE_DL_LIBS)
...
endif()
```